### ISSUE TITLE
[Manually entered locations] No Man's Disconnect - teleporting seems to leave me with someone elses save file

### ISSUE THREAD

**Issue by temp-999** [jaszhix/NoMansConnect#26]
I had version .12 and updated (did not uninstall) to .13 tonight. Thought I'd try teleporting to a hand entered address for some colonized hub. Started NMS and appeared on a snow covered planet. Turned around and saw a starship that was not mine. Checked my inventories and none of them was as expected. Neither the crafts or the inventories. Saw an icon for a claimed base and went there. Reloaded previous save file to get back to my base and saved twice.

PC, Windows 10, 64 bit, 1607 current as of June 2017. Use Steam to access NMS.

C:\Users\<me>\AppData\Roaming\HelloGames\NMS\st_##############

C:\Program Files\NoMansConnect

NMC.log:
Sat Jun 24 2017 23:29:13 GMT-0700 (Pacific Daylight Time):    "Initializing No Man's Connect 0.13.0"
Sat Jun 24 2017 23:29:14 GMT-0700 (Pacific Daylight Time):    "Failed to locate NMS install: path doesn't exist."
Sat Jun 24 2017 23:29:16 GMT-0700 (Pacific Daylight Time):    "Current location: -939:-17:698:0:121:1"
Sat Jun 24 2017 23:29:16 GMT-0700 (Pacific Daylight Time):    "Username: temp"
Sat Jun 24 2017 23:29:16 GMT-0700 (Pacific Daylight Time):    "Active save file: C:\\Users\\<me>\\AppData\\Roaming\\HelloGames\\NMS\\st_##############\\storage2.hg"
Sat Jun 24 2017 23:30:21 GMT-0700 (Pacific Daylight Time):    "Initializing No Man's Connect 0.13.0"
Sat Jun 24 2017 23:30:21 GMT-0700 (Pacific Daylight Time):    "Failed to locate NMS install: path doesn't exist."
Sat Jun 24 2017 23:30:22 GMT-0700 (Pacific Daylight Time):    "Username: temp"
Sat Jun 24 2017 23:30:22 GMT-0700 (Pacific Daylight Time):    "Active save file: C:\\Users\\<me>\\AppData\\Roaming\\HelloGames\\NMS\\st_##############\\storage2.hg"
Sat Jun 24 2017 23:30:22 GMT-0700 (Pacific Daylight Time):    "Current location: -939:-17:698:0:121:1"
Sat Jun 24 2017 23:30:54 GMT-0700 (Pacific Daylight Time):    "Initializing No Man's Connect 0.13.0"
Sat Jun 24 2017 23:30:55 GMT-0700 (Pacific Daylight Time):    "Failed to locate NMS install: path doesn't exist."
Sat Jun 24 2017 23:30:56 GMT-0700 (Pacific Daylight Time):    "Username: temp"
Sat Jun 24 2017 23:30:56 GMT-0700 (Pacific Daylight Time):    "Active save file: C:\\Users\\<me>\\AppData\\Roaming\\HelloGames\\NMS\\st_##############\\storage2.hg"
Sat Jun 24 2017 23:30:56 GMT-0700 (Pacific Daylight Time):    "Current location: -939:-17:698:0:121:1"
Sat Jun 24 2017 23:36:53 GMT-0700 (Pacific Daylight Time):    "Initializing No Man's Connect 0.13.0"
Sat Jun 24 2017 23:36:53 GMT-0700 (Pacific Daylight Time):    "Failed to locate NMS install: path doesn't exist."
Sat Jun 24 2017 23:36:54 GMT-0700 (Pacific Daylight Time):    "Current location: -939:-17:698:0:121:1"
Sat Jun 24 2017 23:36:54 GMT-0700 (Pacific Daylight Time):    "Active save file: C:\\Users\\<me>\\AppData\\Roaming\\HelloGames\\NMS\\st_##############\\storage2.hg"
Sat Jun 24 2017 23:36:54 GMT-0700 (Pacific Daylight Time):    "Username: temp"
Sat Jun 24 2017 23:37:34 GMT-0700 (Pacific Daylight Time):    "Initializing No Man's Connect 0.13.0"
Sat Jun 24 2017 23:37:35 GMT-0700 (Pacific Daylight Time):    "Failed to locate NMS install: path doesn't exist."
Sat Jun 24 2017 23:37:36 GMT-0700 (Pacific Daylight Time):    "Username: temp"
Sat Jun 24 2017 23:37:36 GMT-0700 (Pacific Daylight Time):    "Current location: -939:-17:698:0:121:1"
Sat Jun 24 2017 23:37:36 GMT-0700 (Pacific Daylight Time):    "Active save file: C:\\Users\\<me>\\AppData\\Roaming\\HelloGames\\NMS\\st_##############\\storage2.hg"
Sat Jun 24 2017 23:38:17 GMT-0700 (Pacific Daylight Time):    "Initializing No Man's Connect 0.13.0"
Sat Jun 24 2017 23:38:18 GMT-0700 (Pacific Daylight Time):    "Failed to locate NMS install: path doesn't exist."
Sat Jun 24 2017 23:38:18 GMT-0700 (Pacific Daylight Time):    "Username: temp"
Sat Jun 24 2017 23:38:18 GMT-0700 (Pacific Daylight Time):    "Active save file: C:\\Users\\<me>\\AppData\\Roaming\\HelloGames\\NMS\\st_##############\\storage2.hg"
Sat Jun 24 2017 23:38:18 GMT-0700 (Pacific Daylight Time):    "Current location: -939:-17:698:0:121:1"
Sun Jun 25 2017 00:14:41 GMT-0700 (Pacific Daylight Time):    "AJAX Worker failure: fetchRemoteLocations"

Haven't teleported anywhere else but can do so if that will help.

**Comment by Ahjira** [comment]
I'll second this - same thing happens to me when attempting to teleport to an address I manually entered. You don't end up at that location, nor are you "in" your own save file. It's clearly someone else's.

**Comment by jaszhix** [comment]
Its really impossible for the simple reason of other people's save files not being stored on the server, so there's no way for you to be using one other than your own. What is likely happening is an object from the save is being mutated inadvertently before your save file is written. I won't release another version until I'm able to reproduce this.

**Comment by temp-999** [comment]
Don't know if this will help. I saved a copy of the save file after the voxel address was updated. "USN": " is referenced 1396 times.

null string - 2
Mighty Zombie Slayer - 5
neuropuff - 1284
Arokosaki86 - 2
afWngs - 2
ANARCK0S - 1
Brayo - 1
biogecko - 3
blazzing_u - 1
Chainsaw - 1
Chron - 1
coldboxpal - 1
etc.

None of these are me. I started with release .8, I think and always installed over the previous version. Maybe I have an old file of yours? I'd be happy to uninstall and delete the roaming folder too and then re-install if that would benefit you.

**Comment by jaszhix** [comment]
Those values are the player's you've "met" in the game, and since you're probably using the app to teleport you're seeing familiar names. What's not correct is your username not being in the list. Can you email me that save data along with your username?

**Comment by temp-999** [comment]
Sure. How do I email the save game? Also, I've never met anyone in the game.

**Comment by jaszhix** [comment]
Zip up your save directory and send it to jaszhix@gmail.com. That list is of players that have gone near you - so they can appear in your save file even if you don't know about them.

**Comment by temp-999** [comment]
On its way. I only zipped the save file since that's the only altered file that I made a copy of. 

I'm puzzled as to players that have gone near me when I'm using the app in disconnected mode...

**Comment by jaszhix** [comment]
Well, let's be clear, you're referring to the No Man's Sky save file data, correct? I have no control over how Hello Games handles that.

**Comment by temp-999** [comment]
Perhaps I'm unclear. These were my steps.
I started up NMC which I had already set to Disconnect - so maybe NMD. I selected the most recent location from the left side (places that I had visited) but in this case it was a destination that I had manually entered through NMD. I selected Teleport to this destination. At that time I assumed that NMD updated my most recent NMS save file. I started up NMS and found myself in a location that I had not visited before which was expected. I turned around and saw a starship that wasn't mine. At that point I tabbed (playing on a PC) to my inventories and recognized nothing. I saw a marker on the HUD for a base. Went there and saw someone else's base. Interesting but outside my comfort zone. I reloaded the -1 save file and appeared back at my old base. Then I saved twice which forced the NMD altered save file to -2. Exited NMS and saved the oldest save file which I've sent to you.

Now, how can I help?

**Comment by jaszhix** [comment]
Thanks, that does make it easier to understand. As @Ahjira mentioned, he reproduced this by teleporting to a manually entered location - did you do that when this occurred?

**Comment by temp-999** [comment]
Yes. Should I try a location that I'd actually visited before?

**Comment by jaszhix** [comment]
Yeah, try it.

**Comment by temp-999** [comment]
OK. I started up NMD, backed up my NMS save files, selected the second most recent location in NMD and Teleported there. Checked NMS save files and found that the one with the most current timestamp  (storage3.hg) is smaller than the other two. Copied storage3.hg and it appears to be a twin to the storage2.hg file that I sent you. Used HxD to compare the two file - and they're identical.

**Comment by temp-999** [comment]
Found the file. It's in the nmssavetool folder and is called saveCache.json.

**Comment by jaszhix** [comment]
Did the teleport work?

**Comment by jaszhix** [comment]
Did you ...

- [ ] Restore a functional save before this bug ever happened
- [ ] Use it to teleport to a non-manually entered location (from stored locations, not someone else's)
- [ ] Try deleting the junk save slots after restoring the functional save

The game always loads the newest save slot, so if you restore a save slot file (Normal mode is storage.hg - storage_2.hg), you have to clear the ones affected by the bug or it further complicates the issue with unpredictable results.

**Comment by temp-999** [comment]
Hi.

I'm not sure what a functional save is. What I've always done in the past was to reload previous saves from within NMS.

I should have gone to a location that I discovered - this did not happen.

What are junk save slots?

When I use a previous save from within NMS, I make sure to save NMS before I exit the game so that I have a good starting point for the next time that I open NMS.

**Comment by jaszhix** [comment]
Okay, I'm assuming you backed up your saves. Save files are three slots per difficulty. Basically what I'm saying is if you're certain a save slot works, it will still be older than the ones that are corrupt, so you have to delete them. An easier solution is to back up the entire HelloGames directory.

**Comment by temp-999** [comment]
Before I engage in risky behavior, I make a backup of C:\Users\<me>\AppData\Roaming\HelloGames\NMS.

After I recover from a NMD teleport by loading the -1 generation of the NMS save files from within the game, I then save twice. This way I have a good save file for the next time I start up NMS and, in case I need to teleport, I have an addition good copy in the -1 generation of the save files to reload from.

Am I not explaining this well?

**Comment by jaszhix** [comment]
> by loading the -1 generation of the NMS save files from within the game
What do you mean?

**Comment by temp-999** [comment]
NMS create three save files per mode (?). I only play Normal so the save files are storage.json, storage2.json, and storage3.json. The most current save file has the most recent timestamp and can be considered the 0 generation. One generation back ,(-1), would be the slightly older save file and the oldest save file would be generation (-2). My terminology may not apply in this time. I'm a retired mainframe programming who is more conversant in Cobol.

**Comment by jaszhix** [comment]
I sent a follow up email asking you to send a working save file along with a voxel address of a location that failed for you so I can reproduce it.

**Comment by temp-999** [comment]
It's on its way.

**Comment by Ahjira** [comment]
Jason let me know if you need anything from me. I’m away from home for awhile today but will be back tonight. My case was similar in that teleporting to a manually stored address - that I had visited personally - loaded what appeaRed to be a completely unique save. Not my ship, not my stats, not my inventory and only 47 million units instead of my 1.08 billion. Not the same planet either... or so it seemed.

**Comment by temp-999** [comment]
I've looked a bit closer at what happens when I select Teleport Here using No Man's Disconnect.

NMD saveCache.json overwrites my current NMS save file.
But my new NMS save file is not identical to the NMD saveCache.json. Comparing these two files, all the changes that I looked at (and there are a lot) had to do with decimal precision. In the majority of the cases, the new NMS save file values had one additional decimal place.  

Tried to attach an image but apparently don't know how. I consistently get: 'Something went really wrong, and we can’t process that file. Try again.'

**Comment by jaszhix** [comment]
@Ahjira Thanks.

Did you launch the game after teleporting? Or reload your save with the game already open? Just starting the game up will create tons of changes to the save file that aren't connected to gameplay so I would be interested in seeing if this is still the case if you do nothing with the game and teleport while the game is closed.

**Comment by temp-999** [comment]
For a consistent approach, please give me the sequential steps that you'd like me to do. I'll do them and then provide a narrative of what occurred. 

Is there a size restriction for attaching images?

**Comment by jaszhix** [comment]
I'm not asking for anything specific. I'm just saying it's likely caused by the game not the app. You can confirm this by doing what you did without launching the game. 

I would prefer you use a service like pastebin, images are not  needed for JSON files. I have a bit of work to do on other things before I can get to the app and test it. I'll to try to reproduce it tomorrow.

**Comment by temp-999** [comment]
OK. I'll make a copy of my current NMS save file. Start NMD. Pick a destination and 'Teleport Here'. Then compare saveCache.json to the new NMS current save file. NMS won't be started at all.

Maybe Imgur?  https://imgur.com/a/tsH3k

**Comment by jaszhix** [comment]
No images. send the files to me please.

**Comment by temp-999** [comment]
Sure. Right after I re-run the teleport.

**Comment by temp-999** [comment]
saveCache.json in the NMC folder is  dated 6/24/2017 10:16 Pm and is 1,387 KB.
My current (most recent) NMS save file is storage2.hg  is dated 6/30/2017 2:49 AM and is 2,019 KB.

Saved a copy of saveCache.json.
Saved a copy of storage2.hg.

Started No Man's Disconnect (NMD) and clicked on one of the locations that I had discovered.  Voxel address is -1007:-59:614:0:306:0 .
When the box for that location popped up in the bottom left of the NMD screen I clicked on the three dots (...) and then clicked on Teleport Here.
Saved a copy of storage2.hg as storage2_after_teleport.hg which now is dated 6/30/2017 6:33 PM and is 1,388 KB.

Compared saveCache.json and storage2_after_teleport.json. Differences detected seem to be due to decimal precision. 

Manually reverted my NMS storage2.hg file by copying the originally saved version over top of it.

As a non-Pro, Pastebin on allows 512 KB. Probably fortunate that I don't know how to segment files.

For the next six hours or so zipped copies to saveCache.json and storage2_after_teleport.hg can be found at  http://dropcanvas.com/PD6KBaYX1efm09 .  If that doesn't work, I'll email them to you.

**Comment by Ahjira** [comment]
Ok Jason -- success -- teleporting while the game is open works as you mentioned. I'm not sure how I did it before. Will test with a few other locations. @temp-999  did you still have issues after loading the coordinates this way?

**Comment by temp-999** [comment]
@Ahjira 

Just now:
Started NMS and was at my base where I should be.
Started NMD (NMC, Options, Offline Mode: already set to On), selected -1007:-59:614:0:306:0 in Euclid and Teleported.
My current NMS save file was replaced by C:\Program Files\NoMansConnect\nmssavetool\saveCache.json.
Back in NMS, reloaded current and found myself at -1749:2:-24:1:31:2 in Hilbert.

Are you running Offline? Is this the sequence similar to what you did?

**Comment by Ahjira** [comment]
No I'm running in online mode and I was successful when I teleported exactly how Jason mentioned. Start the game and the pause it by going into your Options screen. Then open or go to NMC and teleport to the address you want. Flip back to NMS and on that same Options screen, immediately reload your current save. This will load up the altered save file that NMC just altered when you opted to teleport and it'll place you out in your ship by the space station of the system you chose.

**Comment by jaszhix** [comment]
@temp-999 Please try v0.14.0 and see if it helps.

**Comment by temp-999** [comment]
Thank you for your continued work!

Did not start NMS.

Downloaded NMC .14.
Installed over version .13. Blocked by firewall. Unblocked and re-installed: OK.

Started NMC.
Picked a location that I'd been to:  -1007:-59:614:0:306:0
In the box that popped-up, selected ... and teleported.
New Stored Location entry appeared in list:  678:-2:1516:0:10:3

Storage2.hg size was changed from 2,019 KB to 1,323 KB.
C:\Program Files\NoMansConnect\nmssavetool\saveCache.json is 1,322 KB.
	"PlayerStateData": {
		"UniverseAddress": {
			"RealityIndex": 0,
			"GalacticAddress": {
				"VoxelX": 678,
				"VoxelY": -2,
				"VoxelZ": 1516,
				"SolarSystemIndex": 10,
				"PlanetIndex": 3

NMC.log for Jul 08 2017:
Sat Jul 08 2017 12:42:14 GMT-0700 (Pacific Daylight Time):    "Initializing No Man's Connect 0.14.0"
Sat Jul 08 2017 12:42:15 GMT-0700 (Pacific Daylight Time):    "Failed to locate NMS install: path doesn't exist."
Sat Jul 08 2017 12:42:17 GMT-0700 (Pacific Daylight Time):    "Active save file: C:\\Users\\<me>\\AppData\\Roaming\\HelloGames\\NMS\\st_76561198053233295\\storage2.hg"
Sat Jul 08 2017 12:42:17 GMT-0700 (Pacific Daylight Time):    "Current location: -939:-17:698:0:121:1"
Sat Jul 08 2017 12:42:17 GMT-0700 (Pacific Daylight Time):    "Username: temp"
Sat Jul 08 2017 12:46:20 GMT-0700 (Pacific Daylight Time):    {"errno":-4048,"code":"EPERM","syscall":"open","path":"C:\\Program Files\\NoMansConnect\\nmssavetool\\saveCache.json"}
Sat Jul 08 2017 15:06:50 GMT-0700 (Pacific Daylight Time):    "Initializing No Man's Connect 0.14.0"
Sat Jul 08 2017 15:06:50 GMT-0700 (Pacific Daylight Time):    "Failed to locate NMS install: path doesn't exist."
Sat Jul 08 2017 15:06:51 GMT-0700 (Pacific Daylight Time):    "Username: temp"
Sat Jul 08 2017 15:06:51 GMT-0700 (Pacific Daylight Time):    "Active save file: C:\\Users\\<me>\\AppData\\Roaming\\HelloGames\\NMS\\st_76561198053233295\\storage2.hg"
Sat Jul 08 2017 15:06:51 GMT-0700 (Pacific Daylight Time):    "Current location: 678:-2:1516:0:10:3"
Sat Jul 08 2017 15:08:02 GMT-0700 (Pacific Daylight Time):    "Initializing No Man's Connect 0.14.0"
Sat Jul 08 2017 15:08:02 GMT-0700 (Pacific Daylight Time):    "Failed to locate NMS install: path doesn't exist."
Sat Jul 08 2017 15:08:03 GMT-0700 (Pacific Daylight Time):    "Current location: 678:-2:1516:0:10:3"
Sat Jul 08 2017 15:08:03 GMT-0700 (Pacific Daylight Time):    "Active save file: C:\\Users\\<me>\\AppData\\Roaming\\HelloGames\\NMS\\st_76561198053233295\\storage2.hg"
Sat Jul 08 2017 15:08:03 GMT-0700 (Pacific Daylight Time):    "Username: temp"

**Comment by jaszhix** [comment]
No problem. So its teleporting you somewhere you've never been to?

**Comment by temp-999** [comment]
That will be correct.

**Comment by jaszhix** [comment]
It looks like the first time you installed the app for all users and the second time you installed it for just your user, and that would suggest it was a permissions issue in your Windows installation the first time you were running into this problem. Thanks for the detailed report.

**Comment by taukausanake** [comment]
I'm having this issue right now and I read over the comments. Is the solution just to reinstall NMC?

**Comment by jaszhix** [comment]
Without any context, I would say open a new issue.
